tracer/test/Datadog.Trace.Tests/Debugger/ProbeExpressionsResources/Approvals/DebuggerExpressionLanguageTests.ToStringNotSupported.verified.txt

```diff
@@ -28,8 +28,43 @@ Expressions:
     var StringArg = (string)scopeMemberArray[9].Value;
     var CollectionArg = (List<string>)scopeMemberArray[10].Value;
     var NestedObjectArg = (DebuggerExpressionLanguageTests.TestStruct.NestedObject)scopeMemberArray[11].Value;
-    var $dd_el_result = "Datadog.Trace.Tests.Debugger.DebuggerExpressionLanguageTests+TestStruct+NestedObject";
+    var $dd_el_result =
+    {
+        var fieldsArray = this.Nested.GetType().GetFields(
+            BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+        var fieldValues = new StringBuilder();
+        var index = 0;
+        while (true)
+        {
+            if ((index < fieldsArray.Length) && (index < 5))
+            {
+                fieldValues.Append(
+                    ProbeExpressionParser<string>.DumpObject(
+                        fieldsArray[index].GetValue(this.Nested),
+                        fieldsArray[index].FieldType,
+                        fieldsArray[index].Name,
+                        0));
+                index++;
+
+                if ((index < fieldsArray.Length) && (index < 5))
+                {
+                    return fieldValues.Append(", ");
+                }
+            }
+            else
+            {
+                if (index < fieldsArray.Length)
+                {
+                    return fieldValues.Append(", ...");
+                }
+
+                break;
+            }
+        }
+
+        return fieldValues.ToString();
+    };
 
     return $dd_el_result;
 }
-Result: The result of the expression is: Datadog.Trace.Tests.Debugger.DebuggerExpressionLanguageTests+TestStruct+NestedObject
+Result: The result of the expression is: _string=I'm a string field, _timeSpan=00:00:00, {[one, 1], [two, 2], [three, 3], ...}, _ienumerable=System.Collections.Generic.IEnumerable`1[[System.Int32, Culture=neutral]], [1, 2, 3, ...], ...
```

Comment:
What happens in case of infinite recursion (i.e. an object that has a field that points to itself, a collection of items that point back at the collection, etc)? Could you please add a test for this?